packages/resolve-scripts/src/server/auth/index.js

```diff
@@ -1,6 +1,53 @@
-import { getRouteByName } from './helper'
-import localStrategy from './localStrategy'
-import githubStrategy from './githubStrategy'
-import googleStrategy from './googleStrategy'
+import localStrategyCallbackCreator from './strategy-callback-creators/local'
+import azureadStrategyCallbackCreator from './strategy-callback-creators/azuread'
+import oauthStrategyCallbackCreator from './strategy-callback-creators/oauth'
 
-export { getRouteByName, localStrategy, githubStrategy, googleStrategy }
+const strategyCallbackCreators = {
+  local: localStrategyCallbackCreator,
+  'azuread-openidconnect': azureadStrategyCallbackCreator,
+  github: oauthStrategyCallbackCreator,
+  google: oauthStrategyCallbackCreator
+}
+
+const getStarategyName = (PassportStrategy, options) => {
+  const strategy = new PassportStrategy(
+    {
+      ...options,
+      passReqToCallback: true
+    },
+    () => {}
+  )
+  return strategy.name
+}
+
+export default (PassportStrategy, options) => {
+  const strategyName = getStarategyName(PassportStrategy, options.strategy)
+
+  const callbackCreator = strategyCallbackCreators[strategyName]
+  if (!callbackCreator) {
+    throw new Error(`Callback for the '${strategyName}' strategy is absent`)
+  }
+
+  return Object.keys(options.routes).map(key => {
```

Comment:
you can write it shorter:

Object.keys(options.routes).map(key => ({
  route: options.routes[key],
  ....
})